File: sdk/ark-sdk-py/ark_sdk/http.py
mport httpx
from .types import ArkError


async def http_async(method: str, url: str, json=None, headers=None):
async with httpx.AsyncClient() as client:
res = await client.request(method, url, json=json, headers=headers)
if res.status_code >= 400:
raise ArkError(res.status_code, res.text)
return res.json() if res.text else None


def http(method: str, url: str, json=None, headers=None):
res = httpx.request(method, url, json=json, headers=headers)
if res.status_code >= 400:
raise ArkError(res.status_code, res.text)
return res.json() if res.text else None

File: sdk/ark-sdk-js/src/http.ts
import { ArkError } from "./types";


export async function http<T>(
method: "GET" | "POST",
url: string,
body?: unknown,
headers: Record<string, string> = {}
): Promise<T> {
const res = await fetch(url, {
method,
headers: {
"content-type": "application/json",
...headers,
},
body: body ? JSON.stringify(body) : undefined,
});


if (!res.ok) {
const msg = await safeText(res);
const err: ArkError = {
status: res.status,
message: msg || `Request failed: ${res.status}`,
};
throw err;
}


const text = await res.text();
return text ? (JSON.parse(text) as T) : (undefined as unknown as T);
}


async function safeText(res: Response): Promise<string | undefined> {
try {
return await res.text();
} catch {
return undefined;
}
}
